hashring/hashring.go

```diff
@@ -171,7 +177,7 @@ func (r *HashRing) removeReplicasNoLock(server string) {
 	delete(r.serverSet, server)
 	for i := 0; i < r.replicaPoints; i++ {
 		address := fmt.Sprintf("%s%v", server, i)
-		r.tree.Delete(r.hashfunc(address))
+		r.tree.Delete(replicapoint(r.hashfunc(address)))
```

Comment:
That can be done after the hashring abstraction, it is out of scope for this PR.